src/descendPeak.cpp

```diff
@@ -0,0 +1,143 @@
+// Pascal Maas - p.maas@lacdr.leidenuniv.nl
+
+#include <Rcpp.h>
+using namespace Rcpp;
+using namespace std;
+
+double weightedMean(NumericVector mzs, NumericVector ints) {
+    // Get the total intensity
+    double intensity = sum(ints);
+
+    // Calculate the weighted mass
+    int n = ints.length();
+    double total = 0;
+    for (int i = 0; i < n; i++) {
+        total += mzs[i] * ints[i] / intensity;
+    }
+    return total;
+}
```

Comment:
Without testing I would assume that you could reduce the runtime of this function by avoid looping over the vector twice (sum has to go through all indices as well):
```suggestion
double weightedMean(NumericVector mzs, NumericVector ints) {
    int n = ints.length();
    double total = 0;
    double intensity = 0;
    for (int i = 0; i < n; i++) {
        total += mzs[i] * ints[i];
        intensity += ints[i];
    }
    return total / intensity;
}
```

Reply:
I agree, this improves the weighted mean function.